src/Microsoft.Azure.WebJobs.Extensions.Storage/TimeoutHandler.cs

```diff
@@ -42,7 +48,30 @@ public static async Task<T> ExecuteWithTimeout<T>(string operationName, string c
                 // Cancel the Delay.
                 cts.Cancel();
 
-                return await operationTask;
+                try
+                {
+                    // Determine if this was a deadlock. If so, log it and rethrow. Use the passed-in cancellationToken
+                    // to detemrine of this operation was canceled explicitly or was due to an internal network timeout.
+
+                    return await operationTask;
+                }
+                catch (StorageException ex) when (ex.InnerException is OperationCanceledException && !cancellationToken.IsCancellationRequested) // network timeout
+                {
+                    Logger.StorageTimeout(logger, operationName, clientRequestId, sw.ElapsedMilliseconds);
```

Comment:
Most of the other changes were to allow me to plumb a logger through to here so all of this was centralized.